Gems/WhiteBox/Code/Source/WhiteBoxToolApiReflection.cpp

```diff
@@ -81,107 +81,200 @@ namespace WhiteBox
             behaviorContext->Class<WhiteBoxMeshHandle>("WhiteBoxMeshHandle")
                 ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
                 ->Attribute(AZ::Script::Attributes::Module, "whitebox.api")
+                ->Method("IsValid", [](WhiteBoxMeshHandle* whiteBoxMeshHandle) {
+                    return (WhiteBoxMeshFromHandle(*whiteBoxMeshHandle) != nullptr);
+                })
                 ->Method(
                     "InitializeAsUnitCube",
                     [](WhiteBoxMeshHandle* whiteBoxMeshHandle)
                     {
-                        return Api::InitializeAsUnitCube(*WhiteBoxMeshFromHandle(*whiteBoxMeshHandle));
+                        WhiteBoxMesh* whiteboxMesh = WhiteBoxMeshFromHandle(*whiteBoxMeshHandle);
+                        AZ_Assert(whiteboxMesh, "WhiteBoxMesh is not found.");
+                        if (whiteboxMesh)
```

Comment:
update: I'm not sure adding all these extra checks is a good idea. If we don't have a `whiteBoxMesh` then something else has gone horribly wrong... this is just adding a lot of noise. Maybe we just add the assert in `WhiteBoxMeshHandle` to check that `WhiteBoxMeshHandle` isn't `0`? And then put all these callsites back to how they were. What are the conditions where this happens? We want to try and eliminate the error state if we can, not fail silently (but not crash)

Reply:
the assert just show the warning but the application will still crash? is there a way to thrown an exception and let the calling method capture it.

Reply:
`GetWhiteBoxMeshHandle` will return an empty whitebox handle but how is that handled in this logic?

Reply:
What I mean is in this context, to be `GetWhiteBoxMeshHandle` should _never_ return an empty White Box Handle, so the fact it does is a problem we should fix. I think this _should_ be possible 🤔 I hope it is! 😝

Reply:
the component in the test is invalid so there isn't a whitebox mesh to give.

Reply:
Hmm okay, can we make it so the test does have some kind of valid White Box Mesh? Otherwise we can go with this change it just feels like we're then hiding a legit issue that might happen later for other reasons.

Reply:
Might it be possible to stop reaching this point sooner too (in the script that's failing?)

Reply:
there is a call to fetch the whitebox mesh. in python I would expect the API to return `None` but i'm not sure if that is possible in the current API.